### DISCUSSION TITLE
Support exact-sized binary types (`[u8; N]`) in entity generation

### DISCUSSION THREAD

**Discussion by emchristiansen** [SeaQL/sea-orm#1494] [Ideas]
## Motivation

Some client binary types (e.g. cryptographic hashes) are binaries of a fixed size (e.g. `[u8; N]` for some `N`).
For reasons of safety and stack-friendliness, it would be nice for columns defined with [`ColumnDef::binary_len`](https://docs.rs/sea-query/0.28.3/sea_query/table/struct.ColumnDef.html#method.binary_len) to produce struct fields of type `[u8; N]`, not `Vec<u8>` as is currently done.

## Proposed Solutions

Binary columns of fixed size should generate fixed-size binary types during entity generation.

**Comment by siketyan** [comment]
I think there are some ways to achieve this:

## Creating a new ColumnType variant

Creating a new variant such as ColumnType::BinaryFixed or something is the simplest solution I think. Of course you can mix Binary (Vec) and BianryFixed ([u8; N]) .

However this is the knowledge of Rust, not SQL or RDB.

## Adding an option on entity generator

Using this, you will be able to choose to use Vec or slice on generating entities. This can not affect anything of SQL or RDB things, but it is not able to mix Vec or slice in generated code at once.

**Comment by billy1624** [comment]
Hey @emchristiansen, I think we already support "fixed sized" binary type in SeaQuery, `ColumnType::Binary(BlobSize::Blob(Some(10))` https://docs.rs/sea-query/latest/sea_query/table/enum.ColumnType.html.

Next, encoding and decoding of [u8; N] in SeaQuery and SeaORM. I just took a glance at the SQLx docs:
- https://docs.rs/sqlx/latest/sqlx/struct.MySql.html#impl-Type%3CMySql%3E-for-%5Bu8%5D
- https://docs.rs/sqlx/latest/sqlx/struct.Postgres.html#impl-Decode%3C'_,+Postgres%3E-for-%5Bu8;+N%5D
- https://docs.rs/sqlx/latest/sqlx/struct.Sqlite.html#impl-Type%3CSqlite%3E-for-%5Bu8%5D

I think it's doable
